src/AzureRMTools.ts

```diff
@@ -414,8 +414,8 @@ export class AzureRMTools {
                 this.suppressErrorDisplay = true;
 
                 const context: PositionContext = deploymentTemplate.getContextFromDocumentLineAndColumnIndexes(position.line, position.character);
-                if (context.completionItems) {
-                    let completionItemArray: Completion.Item[] = await context.completionItems;
+                if (context.getCompletionItems()) {
```

Comment:
I believe this should be:
```
let completionItemArray: Completion.Item[] = await context.getCompletionItems();
if (completionItemArray) {
```

Or just get rid of the `if` if `completionItemArray` is always defined. Right now you're starting the promise twice

Reply:
The problem with auto-rename, although don't know why the if was there in the first place.